.github/workflows/push-gradle-ci.yml

```diff
@@ -41,19 +41,43 @@ jobs:
     # https://github.com/gradle/actions/blob/main/setup-gradle
     - name: Setup Gradle
       uses: gradle/actions/setup-gradle@v4
+
+    # Run build
     - name: Build with Gradle Wrapper
       run: ./gradlew release
 
+    # Maintain snapshot tag
     - name: Tag snapshot release
       if: ${{ vars.PUBLISH_RELEASE == 'true' }}
       run: |
         # Update the release tag to point to the current commit.
         git tag -f "$RELEASE_TAG"
         git push origin -f "$RELEASE_TAG"
 
+    # Upload an artifact release for later jobs to consume
+    - name: Upload distribution
+      uses: actions/upload-artifact@v4
+      with:
+        name: distribution-artifacts
+        path: build/distributions/
+        if-no-files-found: error
```

Comment:
I'm wondering if we should tag _after_ uploading the artifacts.  Just because the latter seems more likely to fail than the former.

Oh and should the upload have a similar `if` gate?

Reply:
Seems reasonable to move the tag after the upload. I can do that.

The upload could be conditional, but as it is right now the artifacts get associated with the CI build, where GitHub will expire them after (by default) 90 days. I don't see a disadvantage to just storing artifacts for all CI builds.

Reply:
Strictly speaking maybe the tag could even go in a separate job, isolating this job to just the build (which isn't specific to snapshot releases). But having to do a new checkout just in order to tag seems wrong so I'll keep them together for now.